components/telemetry-operator/internal/configbuilder/config_builder.go

```diff
@@ -0,0 +1,82 @@
+package configbuilder
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	telemetryv1alpha1 "github.com/kyma-project/kyma/components/telemetry-operator/apis/telemetry/v1alpha1"
+)
+
+type PipelineConfig struct {
+	InputTag          string
+	MemoryBufferLimit string
+	StorageType       string
+	FsBufferLimit     string
+}
+
+// MergeSectionsConfig merges Fluent Bit filters and outputs to a single Fluent Bit configuration.
+func MergeSectionsConfig(pipeline *telemetryv1alpha1.LogPipeline, pipelineConfig PipelineConfig) (string, error) {
```

Comment:
I totally agree. Maybe we can introduce that in a later point in time since I just wanted to change the internals of this function. There is also a second function `MergeParsersConfig` We can introduce renaming or even the API of how to use them when we touch that function as well?

Reply:
yep, makes sense 👍